Replaced requests with httpx 0.18

grantv9 opened the pull request:
Closes #172

petebryan commented:
Thanks @grantv9!

It looks like there are some functions in your branch still using requests (items in msticpy/data/azure). It would be great to replace all instances if this PR.

Also there are some linting errors coming up. You can find the linters we use here:
https://github.com/microsoft/msticpy/wiki/Contributor-guidelines

Please let us know if you have any questions

grantv9 commented:
You're welcome @petebryan!

I fixed the msticpy/data/azure directory and updated the tests accordingly (using the `respx` library instead of `responses`). The linting issues should also be resolved in the new commit.

grantv9 commented:
Submitted a new PR that fixes the last two pytest errors from test_pkg_imports.py and test_sumologic_driver.py
